test/xds/xds_client_ignore_resource_deletion_test.go

```diff
@@ -0,0 +1,473 @@
+	}
+}
+
+// This helper creates a management server for the test.
+func startManagementServer(t *testing.T) *e2e.ManagementServer {
+	t.Helper()
+	mgmtServer, err := e2e.StartManagementServer(e2e.ManagementServerOptions{})
+	if err != nil {
+		t.Fatalf("Failed to start management server: %v", err)
+	}
+	t.Cleanup(mgmtServer.Stop)
+	return mgmtServer
+}
+
+// This helper generates a custom bootstrap config for the test.
+func generateBootstrapContents(t *testing.T, serverURI string, ignoreResourceDeletion bool) []byte {
+	t.Helper()
+	bootstrapContents, err := bootstrap.Contents(bootstrap.Options{
+		NodeID:                             nodeID,
+		ServerURI:                          serverURI,
+		ServerListenerResourceNameTemplate: e2e.ServerListenerResourceNameTemplate,
+		IgnoreResourceDeletion:             ignoreResourceDeletion,
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	return bootstrapContents
+}
+
+// This helper creates in a bootstrapConfig and .
+func xdsResolverBuilder(t *testing.T, bs []byte) resolver.Builder {
+	t.Helper()
+	resolverBuilder := internal.NewXDSResolverWithConfigForTesting.(func([]byte) (resolver.Builder, error))
+	xdsR, err := resolverBuilder(bs)
+	if err != nil {
+		t.Fatalf("Creating xDS resolver for testing: %v", err)
+	}
+	return xdsR
+}
+
+// This helper creates an xDS-enabled gRPC server using the listener and the
+// bootstrap config passed. This helper then registers the server to testService
+// and returns a func to accept requests.
+func setupGRPCServerWithModeChangeChannel(t *testing.T, bootstrapContents []byte, lis net.Listener) (chan connectivity.ServingMode, func()) {
+	t.Helper()
+	updateCh := make(chan connectivity.ServingMode, 1)
+
+	// Create a server option to get notified about serving mode changes.
+	modeChangeOpt := xds.ServingModeCallback(func(addr net.Addr, args xds.ServingModeChangeArgs) {
+		t.Logf("serving mode for listener %q changed to %q, err: %v", addr.String(), args.Mode, args.Err)
+		updateCh <- args.Mode
+	})
+
+	server := xds.NewGRPCServer(grpc.Creds(insecure.NewCredentials()), modeChangeOpt, xds.BootstrapContentsForTesting(bootstrapContents))
+	t.Cleanup(server.Stop)
+	testpb.RegisterTestServiceServer(server, &testService{})
+
+	return updateCh, func() {
+		if err := server.Serve(lis); err != nil {
+			t.Errorf("Serve() failed: %v", err)
+		}
+	}
+}
+
+// This helper creates a listener resource with using a custom listener created.
+// The test uses this listener to serve the gRPC server that is set up.
+func resourceWithListenerForGRPCServer(t *testing.T) (e2e.UpdateOptions, net.Listener) {
+	t.Helper()
+	lis, err := testutils.LocalTCPListener()
+	if err != nil {
+		t.Fatalf("testutils.LocalTCPListener() failed: %v", err)
+	}
+	t.Cleanup(func() { lis.Close() })
+
+	// Setup the management server to respond with the listener resources.
+	host, port, err := hostPortFromListener(lis)
+	if err != nil {
+		t.Fatalf("failed to retrieve host and port of server: %v", err)
```

Comment:
Nit: s/failed/Failed

Also, include the address of the server in the error string. So, something like this:
`t..Fatalf("Failed to retrieve host and port of listener at %q: %v", lis.Addr(), err)`